.github/workflows/sync_docs.yml

```diff
@@ -82,6 +67,34 @@ jobs:
             ${{ runner.os }}-website-
       - run: yarn install --frozen-lockfile
         working-directory: pantsbuild.org
+
+      # Make a new version, if required
+      - name: Make a new version
+        if: endsWith(inputs.version, "a0")
+        run: |
+          BASE_VERSION=$(echo "${{ inputs.version }}" | sed -E 's/^([0-9]+)\.([0-9]+)\.\w+.*/\1.\2/')
+          # Check if the directory already exists, this might be a re-run.
+          if [ ! -d "versioned_docs/version-$BASE_VERSION" ]; then
+            npm run docusaurus docs:version $BASE_VERSION
+          fi
+        working-directory: pantsbuild.org
+
+      # Calculate destination
+      - name: Calculate destination
+        id: get-destination-dir
+        run: |
+          BASE_VERSION=$(echo "${{ inputs.version }}" | sed -E 's/^([0-9]+)\.([0-9]+)\.\w+.*/\1.\2/')
+          DESTINATION_DIR="versioned_docs/version-$BASE_VERSION"
+          if [ ! -d "pantsbuild.org/$DESTINATION_DIR" ]; then
+            DESTINATION_DIR="docs"
+          fi
```

Comment:
What do you think about merging the steps by changing this `if` to something like the following, so we're not duplicating the computations and conceptually repeating these `if` decisions:

```bash
if [ ! -d "$DESTINATION_DIR" ]; then
    if [[ "${{ inputs.version }}" =~ .*\.a0$ ]]; 
        # a0 => we've cut a new branch from `main`, so make a new version
        npm run docusaurus docs:version $BASE_VERSION
    else
        DESTINATION_DIR=docs
    fi
fi
```

(and change this step to run with `working-directory: pantsbuild.org`)

Reply:
I like it